### PR TITLE
Use IShell interface to get Subsystem pointer, also there was an resource leakage issue with ControllerLink usage (_resourceList.size() != 0)

### PR DESCRIPTION
**Pull request by HaseenaSainul**

### CODE REVIEW

File: MediaSystem.cpp [hunk 1]
```diff
@@ -183,7 +155,7 @@ class WideVine : public IMediaKeys, public widevine::Cdm::IEventListener
         }
 
         if ((config.Certificate.IsSet() == true) && (config.Certificate.Value().empty() == false)) {
-            PluginHost::ISubSystem* subsystem = ControllerLink::Instance().SubSystem();
+            PluginHost::ISubSystem* subsystem = const_cast<WPEFramework::PluginHost::IShell*>(shell)->SubSystems();//ControllerLink::Instance().SubSystem();
```
Comment (sramani-metro) [inline MediaSystem.cpp:158]: Perhaps, an assert for shell?

File: MediaSystem.cpp [hunk 2]
```diff
@@ -32,34 +32,6 @@ using namespace WPEFramework;
 
 namespace CDMi {
 static constexpr const TCHAR ControllerCallsign[] = _T("Controller");
```
Comment (sramani-metro) [inline MediaSystem.cpp:34]: is this static being used other than within the ControllerLink?

Comment (HaseenaSainul) [reply MediaSystem.cpp]: yes, it is redundant, so removed

**Review by sramani-metro — APPROVED** [review]
I will wait for others to comment on the history of this code and whether the code was done like this on purpose. Other wise looks good for me

**Comment by HaseenaSainul** [comment]
Sure :)

**Review by bramoosterhuis — APPROVED** [review]

**Pull request merged**